swift/ql/lib/codeql/swift/elements/expr/ApplyExpr.qll

```diff
@@ -12,3 +28,11 @@ class ApplyExpr extends ApplyExprBase {
     result = "call to ..."
   }
 }
+
+class MethodApplyExpr extends ApplyExpr {
+  MethodApplyExpr() { this.getFunction() instanceof MethodRefExpr }
+
+  AbstractFunctionDecl getMethod() { result = this.getFunction().(MethodRefExpr).getMethod() }
+
+  override Expr getQualifier() { result = this.getFunction().(MethodRefExpr).getBase() }
+}
```

Comment:
No need to repeat the `this.getFunction()(MethodRef)` part over and over:

```suggestion
class MethodApplyExpr extends ApplyExpr {
  MethodRefExpr method;
  
  MethodApplyExpr() { method = this.getFunction() }

  AbstractFunctionDecl getMethod() { result = method.getMethod() }

  override Expr getQualifier() { result = method.getBase() }
}
```

Now a `MethodApplyExpr` has both a `getFunction` and a `getMethod`, and they return different things, right? That seems a bit weird. Is this something we could improve on?

Reply:
maybe renaming `getMethod` to `getMethodDeclaration` would clear up things a bit?

Reply:
Thanks, that look way better!

Maybe renaming `getMethod` to `getMethodDeclaration` would clear up things a bit?

Reply:
> Hm, actually. Since `getStaticTarget` has been specialized for `MethodRefExpr`, do we even need `getMethod` here (except to implement `getStaticTarget`, of course)? Won't `getStaticTarget()` give back the same `AbstractFunctionDecl`?

you are absolutely right!

Reply:
I'd really prefer just killing `getMethod`/`getMethodDeclaration` 😂. Is there a reason you want to keep it around?

Reply:
Awesome 😍!